docs/source/data/data_catalog_yaml_examples.md

```diff
@@ -398,6 +402,78 @@ In this example, Kedro understands that `my_dataframe` is the same dataset in it
 In the pipeline, Kedro uses the `spark.SparkDataset` implementation for saving and `pandas.ParquetDataset`
 for loading, so the first node outputs a `pyspark.sql.DataFrame`, while the second node receives a `pandas.Dataframe`.
 
+### How *not* to use transcoding
+
+Kedro pipelines automatically resolve the node execution order and check to ensure there are no circular dependencies in the pipeline. It is during this process that the transcoded datasets are resolved and the transcoding notation `@...` is stripped. This means within the pipeline the datasets `my_dataframe@spark` and `my_dataframe@pandas` are considered to be one `my_dataframe` dataset. The `DataCatalog`, however, treats transcoded entries as separate datasets, as they are only resolved as part of the pipeline resolution process. This results in differences between your defined pipeline in `pipeline.py` and the resolved pipeline that is run by Kedro, and these differences may lead to unintended behaviours. Thus, it is important to be aware of this when using transcoding.
+
+Below are some examples where transcoding may produce unwanted side effects.
+
+#### Defining a node with the same inputs and outputs
+
+Consider the following pipeline:
+
+```python
+pipeline(
+    [
+        node(name="my_func1_node", func=my_func1, inputs="my_dataframe@pandas", outputs="my_dataframe@spark"),
+    ]
+)
+```
+
+During the pipeline resolution, the node above is defined as having the dataset `my_dataset` as both its input and output. As a node cannot have the same inputs and outputs, trying to run this pipeline will fail with the following error:
+
+```console
+ValueError: Failed to create node my_func1([my_dataframe@pandas]) -> [my_dataframe@spark].
+A node cannot have the same inputs and outputs even if they are transcoded: {'my_dataframe'}
+```
+
+#### Defining several nodes that share the same output
```

Comment:
Is this possible too? I think this should not be allowed at all with pipeline validation.

Reply:
No, not possible, same as above.